pkg/kubelet/kubelet.go

```diff
@@ -2822,6 +2822,10 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) (*v1.Pod, error) {
 		if err := kl.statusManager.SetPodAllocation(pod); err != nil {
 			return nil, err
 		}
+		for _, container := range pod.Spec.Containers {
+			key := kuberuntime.GetStableKey(pod, &container)
+			kl.backOff.Reset(key)
+		}
```

Comment:
I think we should only reset the backoff if the container is being resized? Can you think of any reason to reset the others?
```suggestion
		for i, container := range pod.Spec.Containers {
			if !apiequality.Semantic.DeepEqual(container.Resources, allocatedPod.Spec.Containers[i].Resources) {
				key := kuberuntime.GetStableKey(pod, &container)
				kl.backOff.Reset(key)
			}
		}
```

Reply:
True, Done. Code is cleaner now, thanks for sharing the DeepEqual method.